userprefs.php

```diff
@@ -122,13 +122,9 @@
     // of checkboxes at a time.
     // First parameter: true/false.
     // Following parameters: The name of the checkboxes.
-    // The code checks that a checkbox really exists
-    // before accessing it.
     function check_boxes(value) {
-        var f = document.forms[0];
         for (var i = 1; i < arguments.length; i++) {
-            var name = arguments[i];
-            eval('if (f.'+name+') f.'+name+'.checked=value');
+            document.getElementsByName(arguments[i])[0].checked = value;
```

Comment:
We should check that the box exists. This function is used in _show_credits_wanted_adhoc() where pm_credit might not exist and causes a TypeError exception when changing all the check boxes.

Reply:
Ah, good catch - I missed that the PM checkbox was conditional. I've fixed it.